addons/binding/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/internal/VerisureBindingConstants.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.verisure.internal;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+
+/**
+ * The {@link VerisureBinding} class defines common constants, which are
+ * used across the whole binding.
+ *
+ * @author l3rum - Initial contribution
+ */
+@NonNullByDefault
+public class VerisureBindingConstants {
+
+    public static final String BINDING_ID = "verisure";
+
+    // List of all Thing Type UIDs
+    public static final ThingTypeUID THING_TYPE_BRIDGE = new ThingTypeUID(BINDING_ID, "bridge");
+    public static final ThingTypeUID THING_TYPE_ALARM = new ThingTypeUID(BINDING_ID, "alarm");
+    public static final ThingTypeUID THING_TYPE_SMARTPLUG = new ThingTypeUID(BINDING_ID, "smartPlug");
+    public static final ThingTypeUID THING_TYPE_SMOKEDETECTOR = new ThingTypeUID(BINDING_ID, "smokeDetector");
+    public static final ThingTypeUID THING_TYPE_WATERDETETOR = new ThingTypeUID(BINDING_ID, "waterDetector");
```

Comment:
Just spotted this typo "WATERDETETOR" and thought I might as well nitpick it :)

Reply:
Good spotting! I've fixed that typo, however I still get the problem when compiling with maven:
````
[ERROR] Code Analysis Tool has found: 
 1 error(s)! 
 0 warning(s) 
 0 info(s)
[ERROR] .binding.verisure.internal.VerisureHandlerFactory.xml:[3]
There were whitespace characters used for indentation. Please use tab characters instead
[INFO] Detailed report can be found at: file:////Users/jannegpriv/git/openhab2-master/git/openhab2-addons/addons/binding/org.openhab.binding.verisure/target/code-analysis/report.html
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 02:04 min
[INFO] Finished at: 2019-04-23T19:26:57+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.openhab.tools.sat:sat-plugin:0.6.1:report (default) on project org.openhab.binding.verisure: 
[ERROR] Code Analysis Tool has found 1 error(s)! 
[ERROR] Please fix the errors and rerun the build. 
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException

````

That file is auto-generated, how can I fix the problem?
I've not updated my branch in the last month or so, so the binding is not upto date with the latest code changes and migrate to bnd.

I need advice on how to proceed!

Reply:
I'm not sure, but I'd check IDE settings regarding indentation first of all.
Many IDEs tend to default to using spaces, sadly.

Reply:
The file is committed in git and that should not be the case. Remove the file, commit the change and than build again.

Reply:
@jannegpriv Is this file generated during build or was it generated once? Also I think it should be placed under `src/main/resources`, not in `src/main/java`.

Reply:
The file is auto-generated during maven build and is placed in `OSGI-INF` folder:

````
$ find . -type f | grep VerisureHandlerFactory.xml
./target/classes/OSGI-INF/org.openhab.binding.verisure.internal.VerisureHandlerFactory.xml
./OSGI-INF/org.openhab.binding.verisure.internal.VerisureHandlerFactory.xml
````

The folder OSGI-INF contains a `.gitignore`with contents `xml.`.
What mechanism re-creates the `org.openhab.binding.verisure.internal.VerisureHandlerFactory.xml`each maven build?

Reply:
@jannegpriv I get this error when I build this pr. So I rebased the pr on the current master and than the error of the OSGI-INF file is gone. So I would suggest to rebase. However, since your previous merges did go cleanly I would suggest to first clean up your pr. 

With the following commands you should be able to fix it:

With git on the command line do the following on your pr branch: 
First fix your pr with rebase interactively: `git rebase -i HEAD~49` 
This should open an editor with all commits on a line with the word `pick` before it. Replace `pick` with `d` for all commits that are not your own and also do this for your merge commit. 
Save the file and exit the editor.

Now you can rebase to the master
With `git pull --rebase origin master` 
This will show merge conflicts.
Fix the conflicts and these conflicted files with `git add .`
Continue with `git rebase --continue`.
When this succeeds you can *force* push the changes back to github with `git push --force-with-lease`

Reply:
OK, I pushed up my latest changes first, I will try to follow your instructions.

Reply:
@Hilbrand The `git rebase -i HEAD~50` (I've added one more commit) seems to have worked fine. However the next step has no effect:
````
$ git pull --rebase origin master
From https://github.com/jannegpriv/openhab2-addons
 * branch                  master     -> FETCH_HEAD
Current branch 4788-verisure is up to date.
jannegpriv in /Users/jannegpriv/git/openhab2-master/git/openhab2-addons/addons/binding/org.openhab.binding.verisure (4788-verisure) (13 entries, 2 hidden)
````
My remotes:
````
$ git remote -v
origin	https://github.com/jannegpriv/openhab2-addons.git (fetch)
origin	https://github.com/jannegpriv/openhab2-addons.git (push)
upstream	https://github.com/openhab/openhab2-addons.git (fetch)
upstream	https://github.com/openhab/openhab2-addons.git (push)
````

Should that command be `git pull --rebase upstream master` in my 4788-verisure branch?

As I wrote earlier I usually sync my local branch towards upstream master using these steps ([] is indicating local branch):
1. `git fetch upstream`
2. `git checkout master`
3. `[master] git merge upstream/master`
4. `git checkout local_feature_branch`
5. `[local_feature_branch] git merge master`

In this case the local_feature_branch is 4788-verisure branch.

Could I use those steps instead of the `git pull --rebase upstream master`  so I get my local master branch up to date as well?

Reply:
Ah ok. I missed your comment about your syncing. The steps I described are when you use the remote as standard installed by Eclipse with openHAB. 

Assuming you don't change anything on your master you should use for step 5:
5. `[local_feature_branch] git rebase master`

There are of course several ways using git. But we don't use merge commits so step 5 should do a rebase instead of a merge. This keeps it a little bit simpler in general. As a consequence when pushing this rebase you *must* use force with git. `git push --force-with-lease` (or `--force`. But I prefer to recommend `--force-with-lease` as it's the safe force option`.

Reply:
@Hilbrand It seems to have gone well, and now my local branch compiles again!
I've not yet pushed up the changes to my upstream branch since I want to perform some local tests first. 
One question  a lot of bindings and other files/directories are listed as untracked, is that anticipated? I've not listed all untracked bindings below:

````
$ git status
On branch 4788-verisure
Your branch and 'origin/4788-verisure' have diverged,
and have 374 and 50 different commits each, respectively.
  (use "git pull" to merge the remote branch into yours)

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	../org.openhab.binding.airquality/
	../org.openhab.binding.airvisualnode/
	../org.openhab.binding.allplay/
	../org.openhab.binding.amazondashbutton/
	../org.openhab.binding.amazonechocontrol/
	../org.openhab.binding.astro/
        ......
	../org.openhab.binding.zway/
	../../extensionservice/
	../../io/org.openhab.io.azureiothub/
	../../io/org.openhab.io.imperihome/
	../../io/org.openhab.io.javasound/
	../../io/org.openhab.io.mqttembeddedbroker/
	../../io/org.openhab.io.neeo/
	../../io/org.openhab.io.openhabcloud/
	../../io/org.openhab.io.rest.docs/
	../../io/org.openhab.io.transport.modbus/
	../../io/org.openhab.io.webaudio/
	../../persistence/
	../../transform/
	../../ui/
	../../voice/
````

Reply:
Yes that is anticipated. These are generated files that are left behind after the rebase because those bindings have been migrated to the new build. You can remove those files. I'll always do this with `git stash -u` and than `git stash drop`. That removes all untracked files.

Reply:
Pushed up my changes, thanks for the help @Hilbrand! :-)
I also updated the link to the newly built jar-file in my original post.
What is the next step? 
Should the binding be adapted to new build system?

Reply:
Changed title to WIP since Verisure now has launched a complete new Web interface, it will take time to reverse engineer the new API so this binding needs quite some work to get useful again.

Reply:
It looks like the new API is based on GraphQL, do we have any support for GraphQL in some Java library in OpenHAB?

Reply:
I've updated the binding to the new bnd environment and also started the work to transition to the new Verisiure API. 

All set API calls uses new endpoints, most of the get calls uses the old endpoints which are still working.

I will publish a new jar-file for tests and continue to adapt the binding to the new API for all get calls as well.

Reply:
I now completed the adaptations to the new Verisure API, new jar-file has been published to some beta-testers and after successful feed-back  that I will publish it on the [community thread.](https://community.openhab.org/t/testers-for-verisure-openhab-2-binding/7499/134?u=jannegpriv)

I removed the WIP since the binding is once more ready for review.

Reply:
Updated to handle two different API servers.

Reply:
Removed WIP, ready for code review.